src/nsolid.h

```diff
@@ -1764,6 +1815,13 @@ void on_configuration_hook_proxy_(std::string info, void* data) {
   (*static_cast<G*>(data))(info);
 }
 
+template <typename G>
+void on_log_write_hook_proxy_(SharedEnvInst inst,
+                              LogWriteInfo info,
+                              void* data) {
+  (*static_cast<G*>(data))(inst, info);
```

Comment:
I forgot to add... do you think it's possible to avoid making 2 copies of `LogWriteInfo` on every log?